itest/lnd_multi-hop_test.go

```diff
@@ -1838,3 +1838,291 @@ func createThreeHopNetwork(ht *lntest.HarnessTest,
+
+	// Now that we've created the invoice, we'll send a single payment from
+	// Alice to Carol. We won't wait for the response however, as Carol
+	// will not immediately settle the payment.
+	req := &routerrpc.SendPaymentRequest{
+		PaymentRequest: eveInvoice.PaymentRequest,
+		TimeoutSeconds: 60,
+		FeeLimitMsat:   noFeeLimitMsat,
+	}
+	alice.RPC.SendPayment(req)
+
+	// Once the payment sent, Alice should have one outgoing HTLC active.
+	ht.AssertOutgoingHTLCActive(alice, aliceChanPoint, payHash[:])
+
+	// Bob should have two HTLCs active. One incoming HTLC from Alice, and
+	// one outgoing to Carol.
+	ht.AssertIncomingHTLCActive(bob, aliceChanPoint, payHash[:])
+	htlc := ht.AssertOutgoingHTLCActive(bob, bobChanPoint, payHash[:])
+
+	// Carol should have one incoming HTLC from Bob.
+	ht.AssertIncomingHTLCActive(carol, bobChanPoint, payHash[:])
+
+	// Wait for Carol to mark invoice as accepted. There is a small gap to
+	// bridge between adding the htlc to the channel and executing the exit
+	// hop logic.
+	ht.AssertInvoiceState(stream, lnrpc.Invoice_ACCEPTED)
+
+	// Bob now goes offline so the link between Bob and Carol is broken.
+	restartBob := ht.SuspendNode(bob)
+
+	// Carol now settles the invoice, since her link with Bob is broken,
+	// Bob won't know the preimage.
+	carol.RPC.SettleInvoice(preimage[:])
+
+	// We'll now mine enough blocks to trigger Carol's broadcast of her
+	// commitment transaction due to the fact that the HTLC is about to
+	// timeout. With the default incoming broadcast delta of 10, this
+	// will be the htlc expiry height minus 10.
+	numBlocks := padCLTV(uint32(
+		invoiceReq.CltvExpiry - lncfg.DefaultIncomingBroadcastDelta,
+	))
+	ht.MineBlocks(numBlocks)
+
+	// Carol's force close transaction should now be found in the mempool.
+	// If there are anchors, we also expect Carol's anchor sweep. We now
+	// mine a block to confirm Carol's closing transaction.
+	ht.MineClosingTx(bobChanPoint, c)
+
+	// With the closing transaction confirmed, we should expect Carol's
+	// HTLC success transaction to be broadcast.
+	ht.Miner.AssertNumTxsInMempool(1)
+
+	// Restart Bob. Once he finishes syncing the channel state, he should
+	// notice the force close from Carol.
+	require.NoError(ht, restartBob())
+
+	// Get the current height to compute number of blocks to mine to
+	// trigger the htlc timeout resolver from Bob.
+	_, height := ht.Miner.GetBestBlock()
+
+	// We'll now mine enough blocks to trigger Bob's timeout resolver.
+	numBlocks = htlc.ExpirationHeight - uint32(height) -
+		lncfg.DefaultOutgoingBroadcastDelta
+
+	// Mine empty blocks so Carol's htlc success tx stays in mempool. Once
+	// the height is reached, Bob's timeout resolver will resolve the htlc
+	// by extracing the preimage from the mempool.
+	ht.MineEmptyBlocks(int(numBlocks))
+
+	// For neutrino backend, the timeout resolver needs to extract the
+	// preimage from the blocks.
+	if ht.IsNeutrinoBackend() {
```

Comment:
Alternatively, could to `ht.Skipf` if we detect it's a neutrino backend at the top.

Reply:
agree should be skipped

Reply:
I think we don't have test coverage for this resolver for neutrino tho